compiler/rustc_borrowck/src/type_check/free_region_relations.rs

```diff
@@ -57,7 +57,7 @@ pub(crate) fn create<'tcx>(
     universal_regions: Rc<UniversalRegions<'tcx>>,
     constraints: &mut MirTypeckRegionConstraints<'tcx>,
 ) -> CreateResult<'tcx> {
-    UniversalRegionRelationsBuilder {
+    CreateResultBuilder {
```

Comment:
This name is really objectively worse. The "`CreateResult`" is, afaict, the "**result** of **creat**ing the universal relations". This basically removes all of the *salient* information about what the struct is used for. I think you should revert this change.

Reply:
Did you look at how `CreateResult` is being used? It's immediately destructured into its components -- so what *it's* called is not really that meaningful; however, if you care to rename it, I really think it should get a name that is *more* specific, not *less*. 

The main purpose of this `UniversalRegionRelationsBuilder` is to build the `UniversalRegionRelations` are used throughout borrowck, and the only reason that other data is strung along is because it's convenient.

Reply:
An idea would be to call it `UniversalRegionRelationsBuilderResult`, though that's a bit long.

Reply:
Yeah, I did. I don't like having a type called `FooBuilder` that produces a type not called `Foo`. A deeper clean up is required, I'll do that in a follow-up, and I've removed this commit.